astropy/nddata/nduncertainty.py

```diff
@@ -358,7 +376,204 @@ def _propagate_divide(self, other_uncert, result_data, correlation):
         return None
 
 
-class StdDevUncertainty(NDUncertainty):
+class _VariancePropagationMixin:
+    """
+    Propagation of uncertainties for variances, also used to perform error
+    propagation for variance-like uncertainties (standard deviation and inverse
+    variance).
+    """
+    def _propagate_add_sub(self, other_uncert, result_data, correlation,
+                           subtract=False,
+                           to_variance=lambda x: x, from_variance=lambda x: x):
+        """
+        Error propagation for addition or subtraction of variance or
+        variance-like uncertainties. Uncertainties are calculated using the
+        formulae for variance but can be used for uncertainty convertible to
+        a variance.
+
+        Parameters
+        ----------
+
+        other_uncert : `~astropy.nddata.NDUncertainty` instance
+            The uncertainty, if any, of the other operand.
+
+        result_data : `~astropy.nddata.NDData` instance
+            The results of the operation on the data.
+
+        correlation : float or `numpy.ndarray`-like
+            Correlation of the uncertainties.
+
+        subtract : bool, optional
+            If ``True``, propagate for subtraction, otherwise propagate for
+            addition.
+
+        to_variance : function, optional
+            Function that will transform the input uncertainties to variance.
+            The default assumes the uncertainty is the variance.
+
+        from_variance : function, optional
+            Function that will convert from variance to the input uncertainty.
+            The default assumes the uncertainty is the variance.
+        """
+        if subtract:
+            correlation_sign = -1
+        else:
+            correlation_sign = 1
+
+        if other_uncert.array is not None:
+            # Formula: sigma = dB
+            if other_uncert.unit is not None and (result_data.unit**2 != to_variance(other_uncert.unit)):
```

Comment:
The `result.unit ** 2` is calculated in several places, maybe that could be calculated only once. A good variable name would make this more readable as well. :)

Also the parenthesis are unnecessary here, right?